docs/how-to-guides/how-to-speedup-caching.md

```diff
@@ -0,0 +1,88 @@
+---
+title: How-to speed up computations with caching
+alias:
+    name: speed-up-caching
+    text: How-to speed up computations with caching
+---
+
+pyDVL can cache (memoize) the computation of the utility function
+and speed up some computations for data valuation.
+It is however disabled by default.
+When it is enabled it takes into account the data indices passed as argument
+and the utility function wrapped into the
+[Utility][pydvl.utils.utility.Utility] object. This means that
+care must be taken when reusing the same utility function with different data,
+see the documentation for the [caching package][pydvl.utils.caching] for more
+information.
+
+In general, caching won't play a major role in the computation of Shapley values
+because the probability of sampling the same subset twice, and hence needing
+the same utility function computation, is very low. However, it can be very
+useful when comparing methods that use the same utility function, or when
+running multiple experiments with the same data.
+
+pyDVL supports 3 different caching backends:
+
+- [InMemoryCacheBackend][pydvl.utils.caching.memory.InMemoryCacheBackend]:
```

Comment:
We should add a note that this will be deprecated and changed to shared memory since only wrapping the model into routines that release the GIL can benefit from it.

Reply:
I added a warning next to it.